### PR TITLE
inject git commit information into workflow

### PR DESCRIPTION
**Pull request by rahuldesai1**

### CODE REVIEW

File: latch_cli/centromere/ctx.py [hunk 1]
```diff
@@ -52,7 +53,7 @@ class _CentromereCtx:
     dkr_client: Optional[docker.APIClient] = None
     ssh_client: Optional[paramiko.SSHClient] = None
     pkg_root: Optional[Path] = None  # root
-    disable_auto_version: bool = False
+    version_method: AutoVersionMethod = False
```
Comment (ayushkamat) [inline latch_cli/centromere/ctx.py:56]:
```suggestion
    version_method: AutoVersionMethod = AutoVersionMethod.directory
```

File: latch_cli/centromere/ctx.py [hunk 2]
```diff
@@ -75,11 +76,14 @@ class _CentromereCtx:
...
 
+    git_commit_hash: Optional[str] = None
+    git_is_dirty: bool = False
+
     def __init__(
         self,
         pkg_root: Path,
         *,
-        disable_auto_version: bool = False,
+        version_method: AutoVersionMethod = False,
```
Comment (ayushkamat) [inline latch_cli/centromere/ctx.py:86]: default value is wrong type

File: latch_cli/centromere/ctx.py [hunk 3]
```diff
@@ -298,10 +302,48 @@ def __init__(
                 )
             self.version = self.version.strip()
 
-            if not self.disable_auto_version:
-                hash = hash_directory(self.pkg_root)
-                self.version = f"{self.version}-{hash[:6]}"
-                click.echo(f"  {self.version}\n")
+            from git import GitError, Repo
```
Comment (ayushkamat) [inline latch_cli/centromere/ctx.py:305]: add to setup.py + move import to top level unless its a really hefty import

File: latch_cli/centromere/ctx.py [hunk 4]
```diff
@@ -298,10 +302,48 @@ def __init__(
...
+
+            try:
+                repo = Repo(pkg_root)
+                self.git_commit_hash = repo.head.commit.hexsha
+                self.git_is_dirty = repo.is_dirty()
+            except GitError:
+                pass
+            except Exception as e:
+                click.secho(
+                    "WARN: Exception occured while getting git hash from"
```
Comment (ayushkamat) [inline latch_cli/centromere/ctx.py:315]:
```suggestion
                    "WARN: Exception occurred while getting git hash from"
```

File: latch_cli/workflow_config.py [hunk 5]
```diff
@@ -10,6 +10,12 @@
 from latch_cli.constants import latch_constants
 
 
+class AutoVersionMethod(Enum):
+    directory = "directory"
```
Comment (ayushkamat) [inline latch_cli/workflow_config.py:14]:
```suggestion
    directory_hash = "directory_hash"
```

**Review by ayushkamat — CHANGES_REQUESTED** [review]

File: latch_cli/centromere/ctx.py (hunk 3 above)
Comment (rahuldesai1) [reply latch_cli/centromere/ctx.py]: The import will fail if git is not installed locally, so I am going to import here and catch ImportError

**Pull request merged**